src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -2,33 +2,23 @@
 
 using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Linq;
 using Microsoft.CodeAnalysis.CSharp.Symbols;
 using Microsoft.CodeAnalysis.PooledObjects;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed class BestTypeInferrer
```

Comment:
As we discussed, `BestTypeInferrer` handles arrays, lambdas with multiple returns and conditional, and `MethodTypeInference` handles method invocations. But there is a third implementation of a simple "inference": null-coalescing operator. 
Add PROTOTYPE marker as reminder if there isn't one already.